src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,233 @@
+import { Button } from '@patternfly/react-core/dist/esm/components/Button';
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Modal, ModalVariant } from '@patternfly/react-core/dist/esm/components/Modal';
+import { TextArea } from '@patternfly/react-core/dist/esm/components/TextArea';
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { Stack } from '@patternfly/react-core/dist/esm/layouts/Stack';
+
+import cockpit from 'cockpit';
+import { FormHelper } from 'cockpit-components-form-helper';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { useInit } from 'hooks';
+import { superuser } from 'superuser';
+
+import "./editor.scss";
+import { useFilesContext } from '../app.tsx';
+import { checkFilename } from '../common.ts';
+import { get_owner_candidates } from '../ownership.tsx';
+
+import { edit_file } from './editor.tsx';
+
+const _ = cockpit.gettext;
+
+function read_tag(filename: string): Promise<string> {
+    return new Promise(resolve => {
+        cockpit.file(filename, { superuser: "try" }).read()
+                .then((((_content: string, tag: string) => {
+                    resolve(tag);
+                }) as any /* eslint-disable-line @typescript-eslint/no-explicit-any */));
+    });
+}
+
+async function create_file(filename: string, content?: string, owner?: string) {
+    // let tag = await read_tag(filename);
+    const file = cockpit.file(filename, { superuser: "try" });
+    try {
+        await file.replace("");
+    } catch (err) {
+        console.warn("Cannot create new file", err);
+        throw err;
+    }
+
+    if (owner !== undefined) {
+        try {
+            await cockpit.spawn(["chown", owner, filename], { superuser: "require" });
+        } catch (err) {
+            console.warn("Cannot chown new file", err);
+            throw err;
+        }
+    }
+
+    // re-read the tag as it contains the ownership bits
+    const tag = await read_tag(filename);
+
+    if (content) {
+        // We need to obtain the tag to retain file ownership
+        // Can't do this async because we can't get the tag via await
+        try {
+            await cockpit.file(filename, { superuser: "try" }).replace(content, tag);
+        } catch (err) {
+            console.warn("Cannot set initial file text", err);
+            throw err;
+        }
+    }
+}
+
+export const CreateFileModal = ({ dialogResult, path } : {
+    dialogResult: DialogResult<string>,
+    path: string
+}) => {
+    const [user, setUser] = React.useState<cockpit.UserInfo| undefined>();
+    const [filename, setFilename] = React.useState<string>("");
+    const [owner, setOwner] = React.useState<string | undefined>();
+    const [initial_text, setInitialText] = React.useState<string>("");
```

Comment:
Please don't mix snake_case and camelCase.

Reply:
Agreed, the problem is that we have had a C developer work on the codebase which introduced lowercase everywhere.